src/Parser.php

```diff
@@ -351,13 +352,23 @@ protected function isIpv6Host(string $ipv6): bool
             return false;
         }
 
+        //Only the address block fe80::/10 can have a Zone ID attach to
+        //let's detect the link local significant 10 bits
+        static $address_block = "\xfe\x80";
+
+        if (substr(inet_pton($ipv6) & $address_block, 0, 2) === $address_block) {
+            return true;
+        }
+
         $reducer = function (string $carry, string $char): string {
             return $carry.str_pad(decbin(ord($char)), 8, '0', STR_PAD_LEFT);
         };
 
         $res = array_reduce(str_split(unpack('A16', inet_pton($ipv6))[1]), $reducer, '');
 
-        return substr($res, 0, 10) === self::LOCAL_LINK_PREFIX;
+        static $local_link_prefix = '1111111010';
+
+        return substr($res, 0, 10) === $local_link_prefix;
```

Comment:
You can remove the lines 363-371 and just `return substr(inet_pton($ipv6) & $address_block, 0, 2) === $address_block`.

Reply:
Oh, awesome. Done.